lib/zmq-kernel.js

```diff
@@ -297,7 +297,7 @@ export default class ZMQKernel extends KernelTransport {
   onStdinMessage(message: Message) {
     log("stdin message:", message);
 
-    if (!this.isValidMessage(message)) {
+    if (!this._isValidMessage(message)) {
```

Comment:
`ZMQKernel` still needs some validity checks because it proceeds to do things like `const { msg_type } = message.header;`, which assumes that `message.header` exists.